credentials/tls/certprovider/distributor_test.go

```diff
@@ -141,13 +131,15 @@ func (s) TestDistributor(t *testing.T) {
 	waitAndDo(t, proceedCh, errCh, func() {
```

Comment:
I don't understand this test very well.  Why do you need a separate goroutine vs. doing these operations on the distributor when you would push to `proceedCh`?